From a7790362ac8c79992ce44341c4f55946ca9901cf Mon Sep 17 00:00:00 2001
From: yuanhuihui <yuanhuihui@xiaomi.com>
Date: Thu, 6 Apr 2017 15:41:08 +0800
Subject: [PATCH 235/296] fix incorrect context classloader initialization in
 system_server

not only mPackageName.equals("android") run in system_server,
but also contains the following package:
 - com.android.providers.settings,
 - com.android.server.telecom,
 - com.android.location.fused.

when make application for these package will modify
system_server classloader which is not we want.
---
 core/java/android/app/LoadedApk.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/android/app/LoadedApk.java b/core/java/android/app/LoadedApk.java
index e5711f6307d..57c1255384d 100644
--- a/core/java/android/app/LoadedApk.java
+++ b/core/java/android/app/LoadedApk.java
@@ -973,7 +973,7 @@ public final class LoadedApk {
 
         try {
             java.lang.ClassLoader cl = getClassLoader();
-            if (!mPackageName.equals("android")) {
+            if (!ActivityThread.isSystem()) {
                 Trace.traceBegin(Trace.TRACE_TAG_ACTIVITY_MANAGER,
                         "initializeJavaContextClassLoader");
                 initializeJavaContextClassLoader();
-- 
2.11.0

